mrmustard/lab_dev/bb.py

```diff
@@ -0,0 +1,422 @@
+# Copyright 2024 Xanadu Quantum Technologies Inc.
```

Comment:
Should we name this file something a little more descriptive? branch_and_bound.py?

Reply:
sure